### PR TITLE
Generate centroids with polylabel

### PR DESCRIPTION
**Pull request by dereklieu**
Applies `mapbox/polylabel` to find centroids.

Since `polylabel` doesn't handle `MultiPolygon` types, I first try to combine them using `turf.union`. If the combined feature is still a multipolygon, I just use `turf.centroid`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1042,6 +1045,21 @@ geojson/albers/city-labels.geojson:
 				name:name \
 		> $@
 
+tiles/wapo-2016-election-centroids.mbtiles: geojson/albers/centroid-states.geojson \
+	geojson/albers/centroid-counties.geojson \
+	geojson/albers/centroid-districts.geojson
+	mkdir -p $(dir $@)
+	tippecanoe --projection EPSG:3857 \
+		-f \
+		--named-layer=states:geojson/albers/centroid-states.geojson \
```
Comment (anandthakker) [inline Makefile]: let's name these `state-centroids`, `county-centroids`, `district-centroids` (or something like it) -- the reason being, if we make sure that there are no layer-name collisions, we can use Mapbox's vector-tile "compositing", where they combine VTs from different tilesets on the server side if you use a url like `mapbox://devseed.tileset-one,devseed.tileset-two`.

File: centroids [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      // first, try to combine the multi-polygon into a polygon
+      var combined = turf.polygon(feat.geometry.coordinates[0])
+      feat.geometry.coordinates.slice(1).forEach(function (coord) {
+        combined = turf.union(combined, turf.polygon(coord))
+      })
+
+      // it's non-contiguous, use turf
+      if (combined.geometry.type === 'MultiPolygon') {
+        center = turf.centroid(combined).geometry.coordinates
```
Comment (anandthakker) [inline centroids]: I worry that the actual centroid of a MultiPolygon will often fall outside of the shape.  Instead, could we use `turf-area` on each of the MultiPolygon's constituent polygon parts, reconstruct the largest one into a `Polygon` feature, and apply polylabel to that?

File: Makefile [hunk 3]
```diff
@@ -992,6 +992,9 @@ geojson/albers/state-bounds.json: geojson/albers/states.geojson
 geojson/albers/tile-index.json: geojson/albers/us-10m
 	cat $^/*.geojson | ./reproject-geojson --projection mercator --reverse | node_modules/.bin/tile-index -z 7 -f indexed > $@
 
+geojson/albers/centroid-%: geojson/albers/%
```
Comment (anandthakker) [inline Makefile:995]: Might be faster to use the simplified files `geojson/albers/us-10m/%` for input.

**Review by anandthakker — CHANGES_REQUESTED** [review]
Looks good!  Couple suggestions.

File: Makefile (hunk 1 above)
Comment (dereklieu) [reply Makefile]: Ah didn't know about that, but that makes sense.

Comment (dereklieu) [reply Makefile]: Gonna go with `states-centroids`, `counties-centroids`, `districts-centroids` since the other layers are plural too.

Comment (anandthakker) [reply Makefile]: 👍

File: Makefile (hunk 3 above)
Comment (dereklieu) [reply Makefile:995]: @anandthakker I tried this, but polylabel choked on what appears to be a very simplified polygon that only had three points. Bumping it back up to the non-simplified geometries.

**Comment by dereklieu** [comment]
@anandthakker ok, trying out your idea of using `polylabel` on the largest `polygon` in a `multipolygon` feature. Code is pushed, will update `mapbox://washingtonpost.ds-2016-election-centroids` when it finishes.

**Review by anandthakker — APPROVED** [review]

**Pull request merged**